cwizard.py

```diff
@@ -0,0 +1,124 @@
+"""Module to help the user build a dictionary and choose options"""
+# Author: Daniel Davis
+# Group: CSCK451 Group A
+# Date: 29/09/2023
+from enums import *
+import os
+
+
+class ClientWizard:
+    """Class to gather user options"""
+
+    def __init__(self):
+        self.source = None
+        self.format = None
+        self.dictionary = dict()
+        self.filepath = None

```

Comment:
The requirement is to create a File (like we talked about). So maybe we can:
1 - have a constant with the output file path
2 - ask the user what text to write to file
3 - write the inserted text to file
4 - encrypt it if required
5 - delete the file after we send it to the remote server
Like this:
![Screenshot from 2023-09-29 16-29-59](https://github.com/mr-mdd/CSCK541---EOM---Group-A/assets/102789861/8757823d-0479-468e-8810-fe5afed313eb)

Reply:
It currently uploads a textfile
Can we add the textfile_builder in a later cycle?
I would suggest using the same UUID4 filename generator as is used in the server (client_manager) output.
That way we would not get name conflicts, or risk overwriting an existing file.

Reply:
I think we should use the UUID  for the filename that you have and just actually create a textfile. In Python it is pretty simple, you can adapt something like this:

`def write_input_to_file(filename):
    user_input = input("Please write the text you want to save to the file: ")
    
    with open(filename, 'w') as file:
        file.write(user_input)

# Usage
write_input_to_file("output.txt")`

we would just need to add some condition to check if it should be encrypted before being written to the file

Reply:
I've created another question for whether to upload an existing textfile or build a new one.

I've copied the code for saving the file from the client_manager.
I've created a textfile builder by copying the code for building the dictionary - except it now asks for text line-by-line instead of key and value by item.

I've changed the flow of questions so that the encryption question is asked no matter the data source.

Reply:
Actually, I just noticed that this filename should be generated on the server side. I think on the client side it doesn't matter the filename because we might just delete it after we know it has been sent to the server successfully. We can just have some output or temporary folder where we store the file there until we know it has been correctly sent to the server. After we know it, we can just delete and use the same name and path for all files

Reply:
New version uploaded - have a look and see if this is what you were after.